lib/component-data/reactive-render.js

```diff
@@ -128,25 +127,30 @@ function updateRootDOM(uri, newEl, paths) {
 export function updateDOM(uri, newEl, paths) {
   const isElement = newEl.nodeType === newEl.ELEMENT_NODE,
     isFragment = newEl.nodeType === newEl.DOCUMENT_FRAGMENT_NODE,
-    isDocument = newEl.nodeType === newEl.DOCUMENT_NODE;
+    isDocument = newEl.nodeType === newEl.DOCUMENT_NODE,
+    firstChild = isFragment && newEl.firstElementChild,
+    isFullPageFragment = firstChild && firstChild.tagName === 'META' && firstChild.getAttribute('charset') === 'utf-8';
```

Comment:
Isn't this making the assumption that a `layout`-type component will always start out like this? :
```
<!DOCTYPE html>
<html lang="en" data-uri="{{ default _ref _self }}" data-layout-uri="{{ default _ref _layoutRef }}">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
...
```
but I can imagine someone dumping a `<style>` or `<script>` tag before that `<meta>` tag for a special kind of layout.

Reply:
Hmm, point. Let me see if there's another way to do this.